src/shell/commands/disk_rebalance.cpp

```diff
@@ -0,0 +1,288 @@
+// Copyright (c) 2017, Xiaomi, Inc.  All rights reserved.
+// This source code is licensed under the Apache License Version 2.0, which
+// can be found in the LICENSE file in the root directory of this source tree.
+
+#include "shell/commands.h"
+#include "shell/argh.h"
+
+#include <math.h>
+#include <fmt/ostream.h>
+#include <dsn/utility/errors.h>
+#include <dsn/utility/output_utils.h>
+#include <dsn/utility/string_conv.h>
+#include <dsn/dist/replication/duplication_common.h>
+
+bool validate_cmd(const argh::parser &cmd,
+                  const std::set<std::string> &params,
+                  const std::set<std::string> &flags)
+{
+    if (cmd.size() > 1) {
+        fmt::print(stderr, "too many params!\n");
+        return false;
+    }
+
+    for (const auto &param : cmd.params()) {
+        if (params.find(param.first) == params.end()) {
+            fmt::print(stderr, "unknown param {} = {}\n", param.first, param.second);
+            return false;
+        }
+    }
+
+    for (const auto &flag : cmd.flags()) {
+        if (params.find(flag) != params.end()) {
+            fmt::print(stderr, "missing value of {}\n", flag);
+            return false;
+        }
+
+        if (flags.find(flag) == flags.end()) {
+            fmt::print(stderr, "unknown flag {}\n", flag);
+            return false;
+        }
+    }
+
+    return true;
+}
+
+std::unique_ptr<std::ostream> get_out_stream(const std::string &file_name)
+{
+    auto ostream_ptr = std::unique_ptr<std::ostream>();
+    if (file_name.empty()) {
+        ostream_ptr = std::unique_ptr<std::ostream>(&std::cout);
```

Comment:
I think dsn::ref_ptr is better. For std::cout, you can use add_ref() manually, to avoid the std::cout is deleted.
And if you use unique_ptr here, if some user of get_out_stream forget to call release() for std::cout, it will introduce a disater.

Reply:
https://github.com/XiaoMi/pegasus/pull/498#discussion_r396491993 should be better.